compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -198,10 +198,11 @@ pub trait SubdiagMessageOp<G: EmissionGuarantee> =
 /// `#[derive(LintDiagnostic)]` -- see [rustc_macros::LintDiagnostic].
 #[rustc_diagnostic_item = "LintDiagnostic"]
 pub trait LintDiagnostic<'a, G: EmissionGuarantee> {
-    /// Decorate and emit a lint.
```

Comment:
`decorate_lint` is only responsible for decorating a lint diagnostic. It's not responsible for actually emitting the diagnostic!